src/csharp/Microsoft.Spark.Extensions.Delta/Tables/DeltaTable.cs

```diff
@@ -0,0 +1,290 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using Microsoft.Spark.Interop;
+using Microsoft.Spark.Interop.Ipc;
+using Microsoft.Spark.Sql;
+
+namespace Microsoft.Spark.Extensions.Delta.Tables
+{
+    /// <summary>
+    /// Main class for programmatically interacting with Delta tables.
+    /// You can create DeltaTable instances using the static methods.
+    /// 
+    /// <code>
+    /// DeltaTable.ForPath(sparkSession, pathToTheDeltaTable)
+    /// </code>
+    /// </summary>
+    public class DeltaTable : IJvmObjectReferenceProvider
+    {
+        private readonly JvmObjectReference _jvmObject;
+
+        internal DeltaTable(JvmObjectReference jvmObject)
+        {
+            _jvmObject = jvmObject;
+        }
+
+        JvmObjectReference IJvmObjectReferenceProvider.Reference => _jvmObject;
+
+        /// <summary>
+        /// Create a DeltaTable for the data at the given <c>path</c>.
+        /// 
+        /// Note: This uses the active SparkSession in the current thread to read the table data.
+        /// Hence, this throws error if active SparkSession has not been set, that is,
+        /// <c>SparkSession.GetActiveSession()</c> is empty.
+        /// </summary>
+        /// <param name="path"></param>
+        /// <returns>DeltaTable loaded from the path.</returns>
+        public static DeltaTable ForPath(string path) =>
+            new DeltaTable((JvmObjectReference)SparkEnvironment.JvmBridge.CallStaticJavaMethod(
+                "io.delta.tables.DeltaTable",
+                "forPath",
+                path));
+
+        /// <summary>
+        /// Create a DeltaTable for the data at the given <c>path</c> using the given SparkSession
+        /// to read the data.
+        /// </summary>
+        /// <param name="sparkSession"></param>
+        /// <param name="path"></param>
+        /// <returns>DeltaTable loaded from the path.</returns>
+        public static DeltaTable ForPath(SparkSession sparkSession, string path) =>
+            new DeltaTable((JvmObjectReference)SparkEnvironment.JvmBridge.CallStaticJavaMethod(
+                "io.delta.tables.DeltaTable",
+                "forPath",
+                ((IJvmObjectReferenceProvider)sparkSession).Reference,
```

Comment:
Do we need to specify the `((IJvmObjectReferenceProvider)sparkSession).Reference` ?  Have you tried just passing sparkSession object here ?  I think [PayloadHelper.cs](https://github.com/dotnet/spark/blob/e3ce230ca51122774fde5e451cbbbedea7b94b87/src/csharp/Microsoft.Spark/Interop/Ipc/PayloadHelper.cs#L231) should call .Reference on the sparkSession object for you.

Reply:
I did this because I saw the same convention used in other files.

Reply:
I think I only saw this being used in the [ColumnTests.cs](https://github.com/dotnet/spark/blob/e3ce230ca51122774fde5e451cbbbedea7b94b87/src/csharp/Microsoft.Spark.UnitTest/Sql/ColumnTests.cs#L435)

Reply:
Changed.